FindTeamer/ViewModels/EventViewModel.swift

```diff
@@ -17,17 +18,24 @@ final class EventViewModel {
                   "Tennis",
                   "Chess",
                   "Gym"]
-    
+    @Environment(\.managedObjectContext) var managedObjectContext
+    @FetchRequest(
+        entity: Event.entity(),
+        sortDescriptors: [
+            NSSortDescriptor(keyPath: \Event.eventType, ascending: true)
+        ]
+    )
+    var events: FetchedResults<Event>
     // MARK: - Private properties
     private let eventManager: EventManager
     private var cancellables = Set<AnyCancellable>()
-    
+    var repository:CoreDataService!
     // MARK: - Initializators
-    init(event: EventModel = EventModel(), eventManager: EventManager) {
+    init(event: EventModel = EventModel(), eventManager: EventManager, repository: CoreDataService) {
```

Comment:
Your `CoreDataService` already implements `EventManager`. What's the reason of passing it here?